### ISSUE TITLE
connect() should be invoked with type 'int' as first argument, not 'uint'

### ISSUE THREAD

**Issue by vanti** [ARMmbed/mbed-client-linux#25]
According to the Linux man pages, connect() has the following prototype:

```
int connect(int sockfd, const struct sockaddr *addr,
                   socklen_t addrlen);
```

Yet in m2mconnectionhandlerpimpl.cpp, we are casting the first argument to 'uint', which is not a standard type:
`      connect((uint)_socket_server, (const struct sockaddr *)&_sa_dst, _slen_sa_dst);`

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTCLT-590

**Comment by yogpan01** [comment]
Hi, we will be fixing this issue soon. Thanks for reporting.

**Comment by yogpan01** [comment]
Fixed and released. Can you please check if this is working fine for you ?

**Comment by vanti** [comment]
There is still another instance of connect() call on line 132 with the same issue.

**Comment by yogpan01** [comment]
Sorry missed that one,  this should now fixed and released in version 1.1.6 for mbed-client-linux

**Comment by vanti** [comment]
Thank you. This works for me.
